Automattic/newspack-popups
# feat: new option to show border for inline prompts

### All Submissions:






### Changes proposed in this Pull Request:



Adds a new option for inline and custom placement prompts to display the border wrapping the prompt content on the front-end. This is ON by default, so it won't affect existing prompts.

Closes #483 in a roundabout way. This way, if the prompt contains only a Group block with left or right alignment, turning off the border will allow it to float with the article content in a less obtrusive way:

<img width="795" alt="Screen Shot 2021-04-06 at 6 02 54 PM" src="https://user-images.githubusercontent.com/2230142/113792103-56fbbf80-9702-11eb-9ea1-5cba4865a926.png">

Full-width and wide alignment will still have no visible effect (more or less the same as no alignment), to avoid collision with sidebar elements and other injected elements.

### How to test the changes in this Pull Request:

1. On `master`, create some inline and custom placement prompts.
2. Check out this branch and run `npm run build`.
3. Edit the existing prompts and confirm that a new option appears under Prompt Settings called "Display Prompt Border", which should be enabled by default for existing prompts:

<img width="271" alt="Screen Shot 2021-04-06 at 5 58 42 PM" src="https://user-images.githubusercontent.com/2230142/113791835-bdcca900-9701-11eb-93d2-0807ab6ea2e7.png">

4. Confirm that the option only appears for which are inline or custom placement (not overlay or above-header). Don't update yet because we want to test the default.
5. View the inline and custom placement prompts in an incognito session. Confirm that they appear with a border, as usual.
6. Turn OFF the "Display Prompt Border" option and update. Confirm in the incognito session that these prompts now render without a border or container padding.
7. Create a new prompt and confirm that the "Display Prompt Border" option is ON by default.
8. Publish the prompt and test it on the front-end with the option both on and off.

### Other information:
